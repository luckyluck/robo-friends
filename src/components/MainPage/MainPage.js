import React, { Component } from 'react';

import CardList from '../CardList/CardList';
import SearchBox from '../SearchBox';
import ErrorBoundary from '../ErrorBoundary';
// import Scroll from '../Scroll';

import './MainPage.css';

class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    filterRobots = () => {
        return this.props.robots.filter(
            robot => robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
        );
    };

    render() {
        return (
            <div className="tc">
                <h1 className="f1">RoboFriends</h1>
                <SearchBox change={event => this.props.onSearchChange(event)}/>
                {/*<Scroll>*/}
                <ErrorBoundary>
                    <CardList
                        robots={this.filterRobots()}
                        isPending={this.props.isPending}
                    />
                </ErrorBoundary>
                {/*</Scroll>*/}
            </div>
        );
    }
}

export default MainPage;