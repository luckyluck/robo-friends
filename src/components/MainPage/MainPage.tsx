import * as React from 'react';

import CardList from '../CardList/CardList';
import ErrorBoundary from '../ErrorBoundary';
import SearchBox from '../SearchBox';

import './MainPage.css';

import { IAppProps } from '../../interfaces/app.interface';

class MainPage extends React.Component<IAppProps, any> {
    public componentDidMount() {
        this.props.onRequestRobots();
    }

    public filterRobots = () => {
        return this.props.robots.filter(
            robot => robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
        );
    };

    public render() {
        return (
            <div className="tc">
                <h1 className="f1">RoboFriends</h1>
                <SearchBox change={this.props.onSearchChange}/>
                <ErrorBoundary>
                    <CardList
                        robots={this.filterRobots()}
                        isPending={this.props.isPending}
                    />
                </ErrorBoundary>
            </div>
        );
    }
}

export default MainPage;