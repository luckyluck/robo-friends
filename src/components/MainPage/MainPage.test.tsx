import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';

import MainPage from './MainPage';

describe('MainPage component', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        const mockProps = {
            isPending: false,
            onRequestRobots: jest.fn(),
            onSearchChange: jest.fn(),
            robots: [],
            searchField: ''
        };
        wrapper = shallow(<MainPage {...mockProps} />);
    });

    it('should render component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should filter robots correctly', () => {
        const mockProps2 = {
            isPending: false,
            onRequestRobots: jest.fn(),
            onSearchChange: jest.fn(),
            robots: [{
                email: 'john@gmail.com',
                id: '3',
                name: 'John'
            }],
            searchField: 'John'
        };
        let instance: any = wrapper.instance();
        expect(instance.filterRobots()).toEqual([]);
        const wrapper2 = shallow(<MainPage {...mockProps2} />);
        instance = wrapper2.instance();
        expect(instance.filterRobots()).toEqual(mockProps2.robots);
    });
});