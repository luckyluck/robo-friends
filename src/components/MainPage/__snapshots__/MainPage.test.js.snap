// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage component should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobots={[Function]}
    robots={Array []}
    searchField=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="f1"
>
          RoboFriends
</h1>,
        <SearchBox
          change={[Function]}
/>,
        <ErrorBoundary>
          <Unknown
                    isPending={false}
                    robots={Array []}
          />
</ErrorBoundary>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "RoboFriends",
          "className": "f1",
        },
        "ref": null,
        "rendered": "RoboFriends",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "change": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            isPending={false}
            robots={Array []}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isPending": false,
            "robots": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="f1"
>
            RoboFriends
</h1>,
          <SearchBox
            change={[Function]}
/>,
          <ErrorBoundary>
            <Unknown
                        isPending={false}
                        robots={Array []}
            />
</ErrorBoundary>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "RoboFriends",
            "className": "f1",
          },
          "ref": null,
          "rendered": "RoboFriends",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "change": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              isPending={false}
              robots={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isPending": false,
              "robots": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
